libcontainer/nsenter/nsexec.c

```diff
@@ -344,14 +344,61 @@ static void process_nl_attributes(int pipenum, char *data, int data_size)
 		start += NLA_ALIGN(payload_len);
 	}
 
+	return config;
+}
+
+void nsexec(void)
+{
+	int pipenum;
+
+	// If we dont have init pipe, then just return to the go routine,
+	// we'll only have init pipe when do start or exec
+	pipenum = get_init_pipe();
+	if (pipenum == -1) {
+		return;
+	}
+
+	// Retrieve the netlink header
+	struct nlmsghdr nl_msg_hdr;
+	int		len;
+
+	if ((len = read(pipenum, &nl_msg_hdr, NLMSG_HDRLEN)) != NLMSG_HDRLEN) {
+		pr_perror("Invalid netlink header length %d", len);
+		exit(1);
+	}
+
+	if (nl_msg_hdr.nlmsg_type == NLMSG_ERROR) {
+		pr_perror("Failed to read netlink message");
+		exit(1);
+	}
+
+	if (nl_msg_hdr.nlmsg_type != INIT_MSG) {
+		pr_perror("Unexpected msg type %d", nl_msg_hdr.nlmsg_type);
+		exit(1);
+	}
+
+	// Retrieve data
+	int  nl_total_size = NLMSG_PAYLOAD(&nl_msg_hdr, 0);
+	char data[nl_total_size];
+
+	if ((len = read(pipenum, data, nl_total_size)) != nl_total_size) {
+		pr_perror("Failed to read netlink payload, %d != %d", len,
+			  nl_total_size);
+		exit(1);
+	}
+
+	jmp_buf	env;
+	int	syncpipe[2] = {-1, -1};
+	struct	nsenter_config config = process_nl_attributes(pipenum,
+						data, nl_total_size);
+
 	// required clone_flags to be passed
 	if (config.cloneflags == -1) {
 		pr_perror("Missing clone_flags");
 		exit(1);
 	}
 	// prepare sync pipe between parent and child. We need this to let the
-	// child
-	// know that the parent has finished setting up
+	// child know that the parent has finished setting up
```

Comment:
"We need this to let the child know that the parent has finished setting up..."

it's correct as is :-)